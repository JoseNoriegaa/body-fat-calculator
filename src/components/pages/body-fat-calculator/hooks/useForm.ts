// External dependencies
import { useFormik, yupToFormErrors } from 'formik';
import * as yup from 'yup';

// Types & Interfaces

type TFormParsedValues =
  | {
      isMale: true;
      height: number;
      weight: number;
      waist: number;
      neck: number;
    }
  | {
      isMale: false;
      hip: number;
      height: number;
      weight: number;
      waist: number;
      neck: number;
    };

type TFormValues = {
  isMale: string;
  height: string;
  weight: string;
  hip?: string;
  waist: string;
  neck: string;
};

interface IUseFormProps {
  onSubmit: (values: TFormParsedValues | null) => void;
}

// Constants
const INT_REGEXP = /^(?![0]+)[1-9][0-9]*?$/;
const FLOAT_REGEXP = /^(?![0]+)[1-9][0-9]*(\.[0-9]{1,})?$/;

const formSchema = yup.object().shape({
  isMale: yup.boolean().required('Campo requerido'),
  height: yup
    .string()
    .trim('Campo no válido')
    .matches(INT_REGEXP, 'Ingrese un número válido')
    .required('Campo requerido'),
  neck: yup
    .string()
    .trim('Campo no válido')
    .matches(FLOAT_REGEXP, 'Ingrese un número válido')
    .required('Campo requerido'),
  waist: yup
    .string()
    .trim('Campo no válido')
    .matches(FLOAT_REGEXP, 'Ingrese un número válido')
    .required('Campo requerido'),
  weight: yup
    .string()
    .trim('Campo no válido')
    .matches(FLOAT_REGEXP, 'Ingrese un número válido')
    .required('Campo requerido'),
});

const hipSchema = yup.object().shape({
  hip: yup
    .string()
    .trim('Campo no válido')
    .matches(FLOAT_REGEXP, 'Ingrese un número válido')
    .required('Campo requerido'),
});

const INITIAL_VALUES = {
  isMale: 'true',
  height: '',
  neck: '',
  waist: '',
  hip: '',
  weight: '',
};

function useForm({ onSubmit }: IUseFormProps) {
  const formik = useFormik<TFormValues>({
    initialValues: INITIAL_VALUES,
    validationSchema: formSchema,
    validate: async (values) => {
      try {
        await formSchema.validate(values, { abortEarly: false });
      } catch (error) {
        return yupToFormErrors(error);
      }

      if (values.isMale === 'false' && !values.hip) {
        try {
          await hipSchema.validate(values, { abortEarly: false });
        } catch (error) {
          return yupToFormErrors(error);
        }
      }

      return Promise.resolve();
    },
    onSubmit(values) {
      const parsedValues = {
        isMale: values.isMale === 'true',
        height: parseInt(values.height, 10),
        neck: parseFloat(values.neck),
        waist: parseFloat(values.waist),
        weight: parseFloat(values.weight),
        hip: values.hip ? parseFloat(values.hip) : undefined,
      } as TFormParsedValues;

      onSubmit(parsedValues);
    },
  });

  const clearValues = () => {
    formik.resetForm();

    onSubmit(null);
  };

  return {
    formik,
    clearValues,
  };
}

export type { IUseFormProps, TFormParsedValues };
export default useForm;
