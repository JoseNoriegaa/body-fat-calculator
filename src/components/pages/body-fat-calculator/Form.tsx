/* eslint-disable jsx-a11y/label-has-associated-control */
// External dependencies
import React from 'react';
import clsx from 'clsx';

// Internal dependencies
import TextInput, { Label, ErrorText } from '@/components/shared/TextInput';
import useStyles from './Form.styles';
import useForm, { IUseFormProps } from './hooks/useForm';

// Types & Interfaces

function Form({ onSubmit }: IUseFormProps) {
  const classes = useStyles();

  const { formik, clearValues } = useForm({ onSubmit });

  const isFemale = formik.values.isMale === 'false';

  return (
    <>
      <h1 className={classes.title}>Calculadora de Grasa Corporal</h1>

      <p className={classes.description}>
        El método de la Marian de Estados Unidos (US Navy Method) ofrece una manera sencilla de
        calcular un aproximado del porcentaje de tejido adiposo en el cuerpo de una persona.
      </p>

      <p className={classes.description}>
        Los valores requeridos por la fórmula son los siguientes:
      </p>

      <div className={classes.inputContainer}>
        <Label>Género</Label>

        <div className={classes.groupRadio}>
          <div className={classes.radioContainer}>
            <input
              id="input_man"
              name="isMale"
              type="radio"
              value="true"
              onChange={formik.handleChange}
            />
            <Label htmlFor="input_man">Hombre</Label>
          </div>

          <div className={classes.radioContainer}>
            <input
              id="input_woman"
              name="isMale"
              type="radio"
              value="false"
              onChange={formik.handleChange}
            />
            <Label htmlFor="input_woman">Mujer</Label>
          </div>

          {formik.errors.isMale && (
            <div className={classes.groupRadioError}>
              <ErrorText>{formik.errors.isMale}</ErrorText>
            </div>
          )}
        </div>
      </div>

      <div className={classes.inputContainer}>
        <TextInput
          name="height"
          label="Altura (cm)"
          placeholder="Ingresa tu altura"
          onChange={formik.handleChange}
          value={formik.values.height}
          error={formik.errors.height}
        />
      </div>

      <div className={classes.inputContainer}>
        <TextInput
          name="weight"
          label="Peso (kg)"
          placeholder="Ingresa tu peso"
          onChange={formik.handleChange}
          value={formik.values.weight}
          error={formik.errors.weight}
        />
      </div>

      <div className={classes.inputContainer}>
        <TextInput
          name="waist"
          label="Cintura (cm)"
          placeholder="Ingresa la medida de tu cintura"
          onChange={formik.handleChange}
          value={formik.values.waist}
          error={formik.errors.waist}
        />
      </div>

      {isFemale && (
        <div className={classes.inputContainer}>
          <TextInput
            name="hip"
            label="Cadera (cm)"
            placeholder="Ingresa la medida de tu cadera"
            onChange={formik.handleChange}
            value={formik.values.hip}
            error={formik.errors.hip}
          />
        </div>
      )}

      <div className={classes.inputContainer}>
        <TextInput
          name="neck"
          label="Cuello (cm)"
          placeholder="Ingresa la medida de tu cuello"
          onChange={formik.handleChange}
          value={formik.values.neck}
          error={formik.errors.neck}
        />
      </div>

      <div className={classes.actionsContainer}>
        <button
          type="button"
          className={clsx('reset-button', classes.btn, classes.btnSubmit)}
          // eslint-disable-next-line @typescript-eslint/no-misused-promises
          onClick={formik.submitForm}>
          Calcular
        </button>

        <button
          type="button"
          className={clsx('reset-button', classes.btn, classes.btnClear)}
          onClick={clearValues}>
          Limpiar
        </button>
      </div>
    </>
  );
}

export default Form;
