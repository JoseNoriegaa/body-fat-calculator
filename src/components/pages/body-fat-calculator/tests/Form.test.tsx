// External dependencies
import { fireEvent, render, screen, waitFor } from '@testing-library/react';

// Internal dependencies
import Form from '../Form';
import useForm from '../hooks/useForm';

// Mocks
jest.mock('../hooks/useForm.ts');
const onSubmit = jest.fn();

const useFormMock = useForm as jest.Mock;
const onClearValuesMock = jest.fn();
const submitFormMock = jest.fn();

describe('<Form />', () => {
  beforeEach(() => {
    useFormMock.mockReturnValue({
      formik: {
        values: {},
        errors: {},
        submitForm: submitFormMock,
      },
      clearValues: onClearValuesMock,
    });

    useFormMock.mockClear();
    onClearValuesMock.mockClear();
    submitFormMock.mockClear();
  });

  it('should match snapshot - male', () => {
    const { asFragment } = render(<Form onSubmit={onSubmit} />);

    expect(asFragment()).toMatchSnapshot();
  });

  it('should match snapshot - female', () => {
    useFormMock.mockReturnValue({
      formik: {
        values: { isMale: 'false' },
        errors: {},
        submitForm: submitFormMock,
      },
      clearValues: onClearValuesMock,
    });

    const { asFragment } = render(<Form onSubmit={onSubmit} />);

    expect(asFragment()).toMatchSnapshot();
  });

  it('should match snapshot - male - error state', () => {
    useFormMock.mockReturnValue({
      formik: {
        values: { isMale: 'false' },
        errors: {
          height: 'error',
          weight: 'error',
          waist: 'error',
          neck: 'error',
        },
        submitForm: submitFormMock,
      },
      clearValues: onClearValuesMock,
    });

    const { asFragment } = render(<Form onSubmit={onSubmit} />);

    expect(asFragment()).toMatchSnapshot();
  });

  it('should match snapshot - female - error state', () => {
    useFormMock.mockReturnValue({
      formik: {
        values: { isMale: 'false' },
        errors: {
          hip: 'error',
          height: 'error',
          weight: 'error',
          waist: 'error',
          neck: 'error',
        },
        submitForm: submitFormMock,
      },
      clearValues: onClearValuesMock,
    });

    const { asFragment } = render(<Form onSubmit={onSubmit} />);

    expect(asFragment()).toMatchSnapshot();
  });

  it('should handle submit', async () => {
    render(<Form onSubmit={onSubmit} />);

    const buttons = screen.getAllByRole('button');
    const calcButton = buttons[0];

    expect(calcButton).toHaveTextContent('Calcular');

    fireEvent.click(calcButton);

    await waitFor(() => {
      expect(submitFormMock).toHaveBeenCalled();
    });
  });

  it('should handle clear values', async () => {
    render(<Form onSubmit={onSubmit} />);

    const buttons = screen.getAllByRole('button');
    const calcButton = buttons[1];

    expect(calcButton).toHaveTextContent('Limpiar');

    fireEvent.click(calcButton);

    await waitFor(() => {
      expect(onClearValuesMock).toHaveBeenCalled();
    });
  });
});
