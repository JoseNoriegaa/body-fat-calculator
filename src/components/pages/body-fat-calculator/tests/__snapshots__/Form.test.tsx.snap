// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Form /> should match snapshot - female - error state 1`] = `
<DocumentFragment>
  <h1
    class="title-0-2-1"
  >
    Calculadora de Grasa Corporal
  </h1>
  <p
    class="description-0-2-2"
  >
    El método de la Marian de Estados Unidos (US Navy Method) ofrece una manera sencilla de calcular un aproximado del porcentaje de tejido adiposo en el cuerpo de una persona.
  </p>
  <p
    class="description-0-2-2"
  >
    Los valores requeridos por la fórmula son los siguientes:
  </p>
  <div
    class="inputContainer-0-2-3"
  >
    <label
      class="label-0-2-11"
    >
      Género
    </label>
    <div
      class="groupRadio-0-2-4"
    >
      <div
        class="radioContainer-0-2-6"
      >
        <input
          id="input_man"
          name="isMale"
          type="radio"
          value="true"
        />
        <label
          class="label-0-2-11"
          for="input_man"
        >
          Hombre
        </label>
      </div>
      <div
        class="radioContainer-0-2-6"
      >
        <input
          id="input_woman"
          name="isMale"
          type="radio"
          value="false"
        />
        <label
          class="label-0-2-11"
          for="input_woman"
        >
          Mujer
        </label>
      </div>
    </div>
  </div>
  <div
    class="inputContainer-0-2-3"
  >
    <div
      class="container-0-2-12"
    >
      <label
        class="label-0-2-11"
        for="input_height"
      >
        Altura (cm)
      </label>
      <input
        class="input-0-2-13 inputError-0-2-14"
        id="input_height"
        name="height"
        placeholder="Ingresa tu altura"
        type="text"
        value=""
      />
      <div
        class="errorContainer-0-2-15"
      >
        <p
          class="error-0-2-16"
        >
          error
        </p>
      </div>
    </div>
  </div>
  <div
    class="inputContainer-0-2-3"
  >
    <div
      class="container-0-2-12"
    >
      <label
        class="label-0-2-11"
        for="input_weight"
      >
        Peso (kg)
      </label>
      <input
        class="input-0-2-13 inputError-0-2-14"
        id="input_weight"
        name="weight"
        placeholder="Ingresa tu peso"
        type="text"
        value=""
      />
      <div
        class="errorContainer-0-2-15"
      >
        <p
          class="error-0-2-16"
        >
          error
        </p>
      </div>
    </div>
  </div>
  <div
    class="inputContainer-0-2-3"
  >
    <div
      class="container-0-2-12"
    >
      <label
        class="label-0-2-11"
        for="input_waist"
      >
        Cintura (cm)
      </label>
      <input
        class="input-0-2-13 inputError-0-2-14"
        id="input_waist"
        name="waist"
        placeholder="Ingresa la medida de tu cintura"
        type="text"
        value=""
      />
      <div
        class="errorContainer-0-2-15"
      >
        <p
          class="error-0-2-16"
        >
          error
        </p>
      </div>
    </div>
  </div>
  <div
    class="inputContainer-0-2-3"
  >
    <div
      class="container-0-2-12"
    >
      <label
        class="label-0-2-11"
        for="input_hip"
      >
        Cadera (cm)
      </label>
      <input
        class="input-0-2-13 inputError-0-2-14"
        id="input_hip"
        name="hip"
        placeholder="Ingresa la medida de tu cadera"
        type="text"
        value=""
      />
      <div
        class="errorContainer-0-2-15"
      >
        <p
          class="error-0-2-16"
        >
          error
        </p>
      </div>
    </div>
  </div>
  <div
    class="inputContainer-0-2-3"
  >
    <div
      class="container-0-2-12"
    >
      <label
        class="label-0-2-11"
        for="input_neck"
      >
        Cuello (cm)
      </label>
      <input
        class="input-0-2-13 inputError-0-2-14"
        id="input_neck"
        name="neck"
        placeholder="Ingresa la medida de tu cuello"
        type="text"
        value=""
      />
      <div
        class="errorContainer-0-2-15"
      >
        <p
          class="error-0-2-16"
        >
          error
        </p>
      </div>
    </div>
  </div>
  <div
    class="actionsContainer-0-2-7"
  >
    <button
      class="reset-button btn-0-2-8 btnSubmit-0-2-9"
      type="button"
    >
      Calcular
    </button>
    <button
      class="reset-button btn-0-2-8 btnClear-0-2-10"
      type="button"
    >
      Limpiar
    </button>
  </div>
</DocumentFragment>
`;

exports[`<Form /> should match snapshot - female 1`] = `
<DocumentFragment>
  <h1
    class="title-0-2-1"
  >
    Calculadora de Grasa Corporal
  </h1>
  <p
    class="description-0-2-2"
  >
    El método de la Marian de Estados Unidos (US Navy Method) ofrece una manera sencilla de calcular un aproximado del porcentaje de tejido adiposo en el cuerpo de una persona.
  </p>
  <p
    class="description-0-2-2"
  >
    Los valores requeridos por la fórmula son los siguientes:
  </p>
  <div
    class="inputContainer-0-2-3"
  >
    <label
      class="label-0-2-11"
    >
      Género
    </label>
    <div
      class="groupRadio-0-2-4"
    >
      <div
        class="radioContainer-0-2-6"
      >
        <input
          id="input_man"
          name="isMale"
          type="radio"
          value="true"
        />
        <label
          class="label-0-2-11"
          for="input_man"
        >
          Hombre
        </label>
      </div>
      <div
        class="radioContainer-0-2-6"
      >
        <input
          id="input_woman"
          name="isMale"
          type="radio"
          value="false"
        />
        <label
          class="label-0-2-11"
          for="input_woman"
        >
          Mujer
        </label>
      </div>
    </div>
  </div>
  <div
    class="inputContainer-0-2-3"
  >
    <div
      class="container-0-2-12"
    >
      <label
        class="label-0-2-11"
        for="input_height"
      >
        Altura (cm)
      </label>
      <input
        class="input-0-2-13"
        id="input_height"
        name="height"
        placeholder="Ingresa tu altura"
        type="text"
        value=""
      />
    </div>
  </div>
  <div
    class="inputContainer-0-2-3"
  >
    <div
      class="container-0-2-12"
    >
      <label
        class="label-0-2-11"
        for="input_weight"
      >
        Peso (kg)
      </label>
      <input
        class="input-0-2-13"
        id="input_weight"
        name="weight"
        placeholder="Ingresa tu peso"
        type="text"
        value=""
      />
    </div>
  </div>
  <div
    class="inputContainer-0-2-3"
  >
    <div
      class="container-0-2-12"
    >
      <label
        class="label-0-2-11"
        for="input_waist"
      >
        Cintura (cm)
      </label>
      <input
        class="input-0-2-13"
        id="input_waist"
        name="waist"
        placeholder="Ingresa la medida de tu cintura"
        type="text"
        value=""
      />
    </div>
  </div>
  <div
    class="inputContainer-0-2-3"
  >
    <div
      class="container-0-2-12"
    >
      <label
        class="label-0-2-11"
        for="input_hip"
      >
        Cadera (cm)
      </label>
      <input
        class="input-0-2-13"
        id="input_hip"
        name="hip"
        placeholder="Ingresa la medida de tu cadera"
        type="text"
        value=""
      />
    </div>
  </div>
  <div
    class="inputContainer-0-2-3"
  >
    <div
      class="container-0-2-12"
    >
      <label
        class="label-0-2-11"
        for="input_neck"
      >
        Cuello (cm)
      </label>
      <input
        class="input-0-2-13"
        id="input_neck"
        name="neck"
        placeholder="Ingresa la medida de tu cuello"
        type="text"
        value=""
      />
    </div>
  </div>
  <div
    class="actionsContainer-0-2-7"
  >
    <button
      class="reset-button btn-0-2-8 btnSubmit-0-2-9"
      type="button"
    >
      Calcular
    </button>
    <button
      class="reset-button btn-0-2-8 btnClear-0-2-10"
      type="button"
    >
      Limpiar
    </button>
  </div>
</DocumentFragment>
`;

exports[`<Form /> should match snapshot - male - error state 1`] = `
<DocumentFragment>
  <h1
    class="title-0-2-1"
  >
    Calculadora de Grasa Corporal
  </h1>
  <p
    class="description-0-2-2"
  >
    El método de la Marian de Estados Unidos (US Navy Method) ofrece una manera sencilla de calcular un aproximado del porcentaje de tejido adiposo en el cuerpo de una persona.
  </p>
  <p
    class="description-0-2-2"
  >
    Los valores requeridos por la fórmula son los siguientes:
  </p>
  <div
    class="inputContainer-0-2-3"
  >
    <label
      class="label-0-2-11"
    >
      Género
    </label>
    <div
      class="groupRadio-0-2-4"
    >
      <div
        class="radioContainer-0-2-6"
      >
        <input
          id="input_man"
          name="isMale"
          type="radio"
          value="true"
        />
        <label
          class="label-0-2-11"
          for="input_man"
        >
          Hombre
        </label>
      </div>
      <div
        class="radioContainer-0-2-6"
      >
        <input
          id="input_woman"
          name="isMale"
          type="radio"
          value="false"
        />
        <label
          class="label-0-2-11"
          for="input_woman"
        >
          Mujer
        </label>
      </div>
    </div>
  </div>
  <div
    class="inputContainer-0-2-3"
  >
    <div
      class="container-0-2-12"
    >
      <label
        class="label-0-2-11"
        for="input_height"
      >
        Altura (cm)
      </label>
      <input
        class="input-0-2-13 inputError-0-2-14"
        id="input_height"
        name="height"
        placeholder="Ingresa tu altura"
        type="text"
        value=""
      />
      <div
        class="errorContainer-0-2-15"
      >
        <p
          class="error-0-2-16"
        >
          error
        </p>
      </div>
    </div>
  </div>
  <div
    class="inputContainer-0-2-3"
  >
    <div
      class="container-0-2-12"
    >
      <label
        class="label-0-2-11"
        for="input_weight"
      >
        Peso (kg)
      </label>
      <input
        class="input-0-2-13 inputError-0-2-14"
        id="input_weight"
        name="weight"
        placeholder="Ingresa tu peso"
        type="text"
        value=""
      />
      <div
        class="errorContainer-0-2-15"
      >
        <p
          class="error-0-2-16"
        >
          error
        </p>
      </div>
    </div>
  </div>
  <div
    class="inputContainer-0-2-3"
  >
    <div
      class="container-0-2-12"
    >
      <label
        class="label-0-2-11"
        for="input_waist"
      >
        Cintura (cm)
      </label>
      <input
        class="input-0-2-13 inputError-0-2-14"
        id="input_waist"
        name="waist"
        placeholder="Ingresa la medida de tu cintura"
        type="text"
        value=""
      />
      <div
        class="errorContainer-0-2-15"
      >
        <p
          class="error-0-2-16"
        >
          error
        </p>
      </div>
    </div>
  </div>
  <div
    class="inputContainer-0-2-3"
  >
    <div
      class="container-0-2-12"
    >
      <label
        class="label-0-2-11"
        for="input_hip"
      >
        Cadera (cm)
      </label>
      <input
        class="input-0-2-13"
        id="input_hip"
        name="hip"
        placeholder="Ingresa la medida de tu cadera"
        type="text"
        value=""
      />
    </div>
  </div>
  <div
    class="inputContainer-0-2-3"
  >
    <div
      class="container-0-2-12"
    >
      <label
        class="label-0-2-11"
        for="input_neck"
      >
        Cuello (cm)
      </label>
      <input
        class="input-0-2-13 inputError-0-2-14"
        id="input_neck"
        name="neck"
        placeholder="Ingresa la medida de tu cuello"
        type="text"
        value=""
      />
      <div
        class="errorContainer-0-2-15"
      >
        <p
          class="error-0-2-16"
        >
          error
        </p>
      </div>
    </div>
  </div>
  <div
    class="actionsContainer-0-2-7"
  >
    <button
      class="reset-button btn-0-2-8 btnSubmit-0-2-9"
      type="button"
    >
      Calcular
    </button>
    <button
      class="reset-button btn-0-2-8 btnClear-0-2-10"
      type="button"
    >
      Limpiar
    </button>
  </div>
</DocumentFragment>
`;

exports[`<Form /> should match snapshot - male 1`] = `
<DocumentFragment>
  <h1
    class="title-0-2-1"
  >
    Calculadora de Grasa Corporal
  </h1>
  <p
    class="description-0-2-2"
  >
    El método de la Marian de Estados Unidos (US Navy Method) ofrece una manera sencilla de calcular un aproximado del porcentaje de tejido adiposo en el cuerpo de una persona.
  </p>
  <p
    class="description-0-2-2"
  >
    Los valores requeridos por la fórmula son los siguientes:
  </p>
  <div
    class="inputContainer-0-2-3"
  >
    <label
      class="label-0-2-11"
    >
      Género
    </label>
    <div
      class="groupRadio-0-2-4"
    >
      <div
        class="radioContainer-0-2-6"
      >
        <input
          id="input_man"
          name="isMale"
          type="radio"
          value="true"
        />
        <label
          class="label-0-2-11"
          for="input_man"
        >
          Hombre
        </label>
      </div>
      <div
        class="radioContainer-0-2-6"
      >
        <input
          id="input_woman"
          name="isMale"
          type="radio"
          value="false"
        />
        <label
          class="label-0-2-11"
          for="input_woman"
        >
          Mujer
        </label>
      </div>
    </div>
  </div>
  <div
    class="inputContainer-0-2-3"
  >
    <div
      class="container-0-2-12"
    >
      <label
        class="label-0-2-11"
        for="input_height"
      >
        Altura (cm)
      </label>
      <input
        class="input-0-2-13"
        id="input_height"
        name="height"
        placeholder="Ingresa tu altura"
        type="text"
        value=""
      />
    </div>
  </div>
  <div
    class="inputContainer-0-2-3"
  >
    <div
      class="container-0-2-12"
    >
      <label
        class="label-0-2-11"
        for="input_weight"
      >
        Peso (kg)
      </label>
      <input
        class="input-0-2-13"
        id="input_weight"
        name="weight"
        placeholder="Ingresa tu peso"
        type="text"
        value=""
      />
    </div>
  </div>
  <div
    class="inputContainer-0-2-3"
  >
    <div
      class="container-0-2-12"
    >
      <label
        class="label-0-2-11"
        for="input_waist"
      >
        Cintura (cm)
      </label>
      <input
        class="input-0-2-13"
        id="input_waist"
        name="waist"
        placeholder="Ingresa la medida de tu cintura"
        type="text"
        value=""
      />
    </div>
  </div>
  <div
    class="inputContainer-0-2-3"
  >
    <div
      class="container-0-2-12"
    >
      <label
        class="label-0-2-11"
        for="input_neck"
      >
        Cuello (cm)
      </label>
      <input
        class="input-0-2-13"
        id="input_neck"
        name="neck"
        placeholder="Ingresa la medida de tu cuello"
        type="text"
        value=""
      />
    </div>
  </div>
  <div
    class="actionsContainer-0-2-7"
  >
    <button
      class="reset-button btn-0-2-8 btnSubmit-0-2-9"
      type="button"
    >
      Calcular
    </button>
    <button
      class="reset-button btn-0-2-8 btnClear-0-2-10"
      type="button"
    >
      Limpiar
    </button>
  </div>
</DocumentFragment>
`;
